camptix.php

```diff
@@ -2933,23 +2923,26 @@ public static function esc_csv( $fields ) {
 		);
 
 		foreach( $fields as $index => $field ) {
-			// Escape trigger characters at the start of a new field
-			$first_cell_character = mb_substr( $field, 0, 1 );
-			$is_trigger_character = in_array( $first_cell_character, $active_content_triggers, true );
-			$is_delimiter         = in_array( $first_cell_character, $delimiters,              true );
 
-			if ( $is_trigger_character || $is_delimiter ) {
-				$field = "'" . $field;
+			if ( ! is_string( $field ) ) {
+				continue;
 			}
 
-			// Escape trigger characters that follow delimiters
-			foreach ( $delimiters as $delimiter ) {
-				foreach ( $active_content_triggers as $trigger ) {
-					$field = str_replace( $delimiter . $trigger, $delimiter . "'" . $trigger, $field );
+			$escaped_field = '';
+
+			// Escape trigger characters that follow delimiters, or are at the start
+			$is_prev_char_delimiter = true;
+			for ( $i = 0; $i < strlen( $field ); $i++ ) {
+				$char = $field[ $i ];
```

Comment:
I think it might be better to name this `$current_char`, to make its purpose more obvious. How does that sound?